### ISSUE TITLE
Forced `throw=True` Behavior in Backward Pass of `lx.linear_solve`

### ISSUE THREAD

**Issue by ASKabalan** [patrick-kidger/lineax#117]
Hello :wave: 

I am using `lineax` for solving a large linear system within a likelihood function. Specifically, I use the CG solver to invert a large matrix.

I am also trying to minimize the likelihood function using some solver from Optax

I set the `throw` parameter in the CG solver to `False`  .. however, in the backward pass the solver can  `throw`

For example
```python
import lineax as lx 


X = ...
solver = lx.NormalCG(rtol=1e-4, atol=1e-4, max_steps=500)

def solve(X):
    SomeOperator = ...


    solution  = lx.linear_solve(SomeOperator, X, solver=solver, throw=False)

    return solution.value


forward = solve(X) # Cannot throw
backward = jax.grad(solve)(X)# Can throw
```
I notices that you documented[ this here](https://github.com/patrick-kidger/lineax/blob/42cf2d8d447007b67a4b1e585c600ed5dfe833fb/lineax/_solve.py#L137)

``` 
# Note that we pass throw=True unconditionally to all the tangent solves, as there
# is nowhere we can pipe their error to.
# This is the primal solve so we can respect the original `throw`.
```

And the `throw=True` is forced [here](https://github.com/patrick-kidger/lineax/blob/42cf2d8d447007b67a4b1e585c600ed5dfe833fb/lineax/_solve.py#L211)

For me it is problematic since in my case I am running the CG solver mutiple times .. and using the gradients to optimize my own function.

For the moment I had to manually comment out the throw function

Do you have any feedback on this?

Thanks

**Comment by patrick-kidger** [comment]
On the forward pass, if we have `sol = linear_solve(..., throw=False)`, then the fact an error was raised appears in `sol.result`, so you can handle it programmaticaly.

But on the backward pass, there's no equivalent. If the solve cannot complete then what is your desired behaviour?

**Comment by ASKabalan** [comment]
Thank you for the fast response

Anything but a crash
Maybe a huge warning via debug.print  to warn the user that his backward pass failed
(I am pretty sure that you can still use debug.callback in the `JVP` implementation right?)

But for me if the Matrix is not inverted with good tolerance, my optimizer will eventually fail to converge but atleast I will get the last value

However if I encapsulate with try catch 1 - it is not very pretty 2 - I won't get the last value which is very usefull

Thanks again

**Comment by patrick-kidger** [comment]
Okay! In that case take a look at the documentation for `equinox.error_if`, which is what is being called here to raise the error. In particular you can set `EQX_ON_ERROR=nan` to have the solution set to a NaN rather than a crash.

**Comment by ASKabalan** [comment]
Thank you very much for your answer :pray:
